contracts/src/mixins/QuotaMixin.sol

```diff
@@ -0,0 +1,200 @@
+        uint16 requiredQuotaMultiplier;
+        uint32 freeQuotaPerDepositedFeeToken;
+        uint32 maxFreeQuota;
+        address feeToken;
+        uint8 feeTokenDecimals;
+    }
+
+    mapping(address => QuotaStats) public quotaStatsForSigner;
+    mapping(address => mapping(bytes32 => uint256)) public withdrawsForSigner;
+
+    uint64 public immutable TIMEFRAME_QUOTA_RESET;
+    uint32 public immutable FREE_QUOTA_PER_DEPOSITED_FEE_TOKEN;
+    uint16 public immutable REQUIRED_QUOTA_MULTIPLIER;
+    uint32 public immutable MAX_FREE_QUOTA;
+    address public immutable FEE_TOKEN;
+    uint8 public immutable FEE_TOKEN_DECIMALS;
+
+    constructor(QuotaMixinConfig memory _config) {
+        TIMEFRAME_QUOTA_RESET = _config.timeframeQuotaReset;
+        FREE_QUOTA_PER_DEPOSITED_FEE_TOKEN = _config
+            .freeQuotaPerDepositedFeeToken;
+        REQUIRED_QUOTA_MULTIPLIER = _config.requiredQuotaMultiplier;
+        MAX_FREE_QUOTA = _config.maxFreeQuota;
+        FEE_TOKEN = _config.feeToken;
+        FEE_TOKEN_DECIMALS = _config.feeTokenDecimals;
+    }
+
+    function depositTokensForSigner(
+        address signer,
+        uint128 amount
+    ) public {
+        // We don't update the nextQuotaReset this way depositing more tokens does not negatively affect the reset schedule
+        // The reset schedule always starts from 0, therefore is always a multiple of the reset timeframe (unless the timeframe is changed)
+        quotaStatsForSigner[signer].tokenBalance += amount;
+        IERC20(FEE_TOKEN).safeTransferFrom(msg.sender, address(this), amount);
+        emit Deposit(signer, amount);
+    }
+
+    function computeWithdrawHash(
+        uint256 amount,
+        address beneficiary,
+        uint256 nonce
+    ) internal view returns (bytes32 withdrawHash) {
+        bytes32 domainSeparator = keccak256(
+            abi.encode(
+                keccak256(
+                    "EIP712Domain(uint256 chainId,address verifyingContract)"
+                ),
+                block.chainid,
+                address(this)
+            )
+        );
+        bytes32 structHash = keccak256(
+            abi.encode(
+                keccak256(
+                    "WithdrawRequest(uint256 amount,address beneficiary,uint256 nonce)"
+                ),
+                amount,
+                beneficiary,
+                nonce
+            )
+        );
+        withdrawHash = keccak256(
+            abi.encodePacked("\x19\x01", domainSeparator, structHash)
+        );
+    }
+
+    function widthdrawTokensForSigner(
+        bytes calldata signature,
+        uint128 amount,
+        address beneficiary,
+        uint256 nonce
+    ) public payable {
+        bytes32 withdrawHash = computeWithdrawHash(amount, beneficiary, nonce);
+        (address signer, bytes32 r, bytes32 vs) = CoreLib.recoverSigner(
+            withdrawHash,
+            signature
+        );
+        // Silence unused local variable warning
+        (r, vs);
```

Comment:
```suggestion
        (address signer,,) = CoreLib.recoverSigner(
            withdrawHash,
            signature
        );
```

I think this should work too.

Reply:
Ahh not defining it at all, will check ... My lack of active coding is showing 😅